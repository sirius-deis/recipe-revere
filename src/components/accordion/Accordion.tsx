import { FC, PropsWithChildren } from "react";
import styles from './Accordion.module.css'

interface AccordionProps {
    title: string;
}

const Accordion: FC<PropsWithChildren & AccordionProps> = ({children, title}) => {
    return <section>
        <button className={styles.accordion}>{title}</button>
        <div className={styles.panel}>
            {children}
        </div>
    </section>
}

export default Accordion;