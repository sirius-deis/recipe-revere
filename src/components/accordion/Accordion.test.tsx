import { render, screen, fireEvent } from "@testing-library/react";
import Accordion from "./Accordion";

const lorem = ` Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero totam magnam cum culpa nemo nesciunt laborum reiciendis necessitatibus ea, pariatur error itaque repudiandae repellat incidunt doloremque earum et officiis quam?`;

describe("Accordion component", () => {
  it("Should match snapshot", () => {
    const { container } = render(
      <Accordion title="Lorem">
        <p>{lorem}</p>
      </Accordion>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render component with proper title and with provided component inside component being hidden", () => {
    render(
      <Accordion title="Lorem">
        <p>{lorem}</p>
      </Accordion>
    );
    expect(screen.getByText("Lorem")).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent("Lorem");
    expect(screen.queryByText(lorem)).not.toBeInTheDocument();
  });
});
