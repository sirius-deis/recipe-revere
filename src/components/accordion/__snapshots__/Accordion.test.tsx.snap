// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Should match snapshot 1`] = `
<div>
  <section
    class="active"
  >
    <button
      class="accordion"
    >
      Lorem
    </button>
    <div
      class="panel"
    >
      <p>
         Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero totam magnam cum culpa nemo nesciunt laborum reiciendis necessitatibus ea, pariatur error itaque repudiandae repellat incidunt doloremque earum et officiis quam?
      </p>
    </div>
  </section>
</div>
`;
